common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -676,16 +680,37 @@ class @Problem
       element = $(element)
       element.find("section[id^='forinput']").removeClass('choicetextgroup_show_correct')
 
+  disableAllButtonsWhileRunning: (buttonsHandlerCallback ,processCallback) =>
+    # Used to keep the buttons disabled while processCallback is running.
+    buttonsHandlerCallback false
+    processCallback().always -> buttonsHandlerCallback true
+
+  enableAllButtons: (enable) =>
+    # Used to enable/disable all buttons in problem.
+    @enableCheckButton enable
+    if enable
+      @resetButton.removeClass('is-disabled').attr({'aria-disabled': 'false'})
+      @saveButton.removeClass('is-disabled').attr({'aria-disabled': 'false'})
+      @hintButton.removeClass('is-disabled').attr({'aria-disabled': 'false'})
+      @showButton.removeClass('is-disabled').attr({'aria-disabled': 'false'})
+    else
+      @resetButton.addClass('is-disabled').attr({'aria-disabled': 'true'})
+      @saveButton.addClass('is-disabled').attr({'aria-disabled': 'true'})
+      @hintButton.addClass('is-disabled').attr({'aria-disabled': 'true'})
+      @showButton.addClass('is-disabled').attr({'aria-disabled': 'true'})
+
   enableCheckButton: (enable) =>
```

Comment:
`@change_check_button_text` should be a parameter in this function. Also, use camelCase for coffeescript.